mono/mini/mini-llvm.c

```diff
@@ -3968,6 +3992,45 @@ process_call (EmitContext *ctx, MonoBasicBlock *bb, LLVMBuilderRef *builder_ref,
 	 */
 	lcall = emit_call (ctx, bb, &builder, callee, args, LLVMCountParamTypes (llvm_sig));
 
+	if (mono_aot_can_specialize (call->method)) {
+		LLVMValueRef lmethod = callee;
+
+		int num_params = LLVMCountParams (lmethod);
+		GArray *call_site_union = (GArray *) g_hash_table_lookup (ctx->module->method_to_call_info, call->method);
+
+		if (!call_site_union) {
+			call_site_union = g_array_sized_new (FALSE, TRUE, sizeof (gboolean), num_params);
```

Comment:
```suggestion
			call_site_union = g_array_sized_new (FALSE, TRUE, sizeof (gint32), num_params);
```

You're using it as a counter, below.

Reply:
Oh yikes, missed that change when I changed the type of this. Good catch.